web/src/engine/websites/XCaliBRScans.ts

```diff
@@ -19,36 +58,106 @@ export default class extends DecoratableMangaScraper {
+        const data = await FetchWindowScript<pageScriptResult>(request, pagescript,2500);
+        // HACK: bypass 'i0.wp.com' image CDN to ensure original images are loaded directly from host
+        const piclist = data.imagz.map(link => link.replace(/\/i\d+\.wp\.com/, '')).filter(link => !link.includes('histats.com'));
+        switch (data.scrambled) {
+
+            case 0:
+            case 1: //Flip each picture, no grouping
+                return piclist.map(pic => {
+                    return new Page(this, chapter, new URL(pic, this.URI), { scrambled: data.scrambled });
+                });
+            case 2: {//Flip and group by 2
+                let count = 0;
+                const pages = [];
+
+                //create one Page for each 2 pictures
+                for (let i = 0; i < piclist.length - 1; i += 2) {
+                    const parameters = { scrambled: data.scrambled, secondaryPic: piclist[i + 1] };
+                    const page = new Page(this, chapter, new URL(piclist[i], this.URI), parameters);
+                    pages.push(page);
+                    count += 2;
+                }
+                //get remaining picture if number was odd
+                if (count < piclist.length) {
+                    pages.push(new Page(this, chapter, new URL(piclist[count], this.URI), { scrambled: 0 }));
+                }
+                return pages;
             }
-        };
+            default:
+        }
     }
 
-    async _getPages(chapter) {
-        const images = await super._getPages(chapter);
-        return images.map(image => this.createConnectorURI(image));
+    public override async FetchImage(page: Page, priority: Priority, signal: AbortSignal): Promise<Blob> {
+        const blobMainImage = await Common.FetchImage.call(this, page, priority, signal);
+        const bitmaps: ImageBitmap[] = [];
+
+        switch (page.Parameters.scrambled) {
+            case 0: return blobMainImage; //No scrambling, return image
+            case 1: //Flip picture
+                bitmaps.push(await createImageBitmap(blobMainImage));
+                return await this.composePuzzle(bitmaps, page.Parameters.scrambled);
+            case 2://Combine/Flip 2 pictures
+            {
+                bitmaps.push(await createImageBitmap(blobMainImage));
+                //fetch second image
+                const pageUrl = (page.Parameters.secondaryPic) as string;
+                const request = new FetchRequest(pageUrl, { referrer: this.URI.href });
+                const response = await Fetch(request);
+                const data = await response.blob();
+                bitmaps.push(await createImageBitmap(data));
+                return await this.composePuzzle(bitmaps, page.Parameters.scrambled);
+            }
+            default :
+        }
+
     }
-}
-*/
\ No newline at end of file
+    async composePuzzle(bitmaps: ImageBitmap[], scrambleMode : number): Promise<Blob> {
+        switch (scrambleMode) {
```

Comment:
Use two different methods to descramble the images, methods with conditional flags should be avoided